packages/core/src/interpreter.ts

```diff
@@ -510,7 +516,15 @@ export class Interpreter<
     }
 
     this.state.configuration.forEach((stateNode) => {
-      for (const action of stateNode.definition.exit) {
+      const exitActions = stateNode.definition.exit;
+      const [resolvedExitActions] = resolveActions(
+        this.machine,
+        this.state,
+        this.state.context,
+        toSCXMLEvent('xstate.stop'), // dummy event, for now
```

Comment:
according to SCXML, this should be `this.state._event` but I actually don't like this part of the SCXML spec 😬

Reply:
Yeah, I had that at first but realized this isn't accurate in real-world: the actions are caused by _stopping_ the state machine (hence `xstate.stop`) rather than by the previous event.